File: leetcode3/이진희/3838. Weighted Word Mapping.java
/* 

1. 아이디어 : 문자열의 배열마다 각 배열 가중치를 센 후 모듈러 연산으로 나눠서 나온값을 알파벳 변환한다

              이때, z=0, a=25 처럼 반대로 계산해야 한다.

2. 시간복잡도 : O(100*10)

3. 자료구조/알고리즘 : 완전탐색

 */

class Solution {
    public String mapWordWeights(String[] words, int[] weights) {
        StringBuilder sb = new StringBuilder();
        for(String s : words) {
            int cnt = 0;
            for(int i=0; i<s.length(); i++) {
                cnt+=weights[s.charAt(i)-'a'];
            }

            char c = (char)('z'-cnt%26);
            sb.append(c);
        }

        return sb.toString();
    }
}

File: leetcode3/변지협/v2/3838. Weighted Word Mapping.py

'''
1. 아이디어 :
alphabet dic, dic_rev 만들고, 이에 매핑해서 구한다.

2. 시간복잡도 :
    o(n * m) n: words 길이, m: word 길이
3. 자료구조/알고리즘 :
'''
class Solution:
    def mapWordWeights(self, words: List[str], weights: List[int]) -> str:
        alphabet = 'abcdefghijklmnopqrstuvwxyz'
        dic = {}
        dic_rev = {}
        wn = len(weights)
        for i in range(wn):
            dic[alphabet[i]] = weights[i]
            dic_rev[25-i] = alphabet[i]
        
        ans = ''
        for word in words:
            tmp = sum([dic[w] for w in word])
            ans += dic_rev[tmp%26]
        
        return ans
        
            